### ISSUE TITLE
[BUG] Refactoring v2prov related conditions

### ISSUE THREAD

**Issue by jakefhyde** [rancher/rancher#41024]
**Rancher Server Setup**
- Rancher version: v2.7-head

**Describe the bug**

Rancher utilizes the `Unknown` condition status to indicate "In Progress", whereas CAPI utilizes the status `False` + a `Severity` field. Additionally the order of condition sorting is as follows:
```
False + Error
False + Warning
False + Info
True
Unknown
Everything else
```
This results in unknown and unexpected results in both CAPI as well as the UI, as CAPI internally as framework is not designed to expect `Unknown` to indicate in progress. Additionally, there is currently a disconnect between satisfying framework contracts, and what leads to a favorable UX. When a cluster is provisioned, the status changes from Provisioning, to Reconciling, to Updating in the UI, and it is unclear what the desired behavior is.

**To Reproduce**

N/A

**Result**

N/A

**Expected Result**

N/A

**Screenshots**

**Additional context**

**Comment by snasovich** [comment]
@Oats87 @jakefhyde , I don't believe we're any longer planning to refactor conditions as part of Q2 release. Kicking it out to Q3, let me know if it's incorrect.

**Comment by snasovich** [comment]
This is unlikely to happen soon so removing milestone.
